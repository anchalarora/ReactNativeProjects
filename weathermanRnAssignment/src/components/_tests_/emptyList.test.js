import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import EmptyList from '../emptyList';

it('should load EmptyList without crashing', () => {
  const rendered = renderer.create(<EmptyList />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it('should render the Text element', () => {
  const component = shallow(<EmptyList label="No Data Found" />);
  expect(component).toBeTruthy();
});

let findElement = function (tree, element) {
  console.log(tree);
  let result = undefined;
  let node;
  for (node in tree.children) {
    if (tree.children[node].props.testID == element) {
      console.log('Inside for loop   ', tree.children[node].props.testID);
      result = true;
    }
  }
  return result;
};

it('find Text element in EmptyList', () => {
  let tree = renderer.create(<EmptyList />).toJSON();
  expect(findElement(tree, 'textItem')).toBeDefined();
});
