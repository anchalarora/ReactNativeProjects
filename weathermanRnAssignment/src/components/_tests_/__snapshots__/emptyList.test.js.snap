// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should load EmptyList without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "marginEnd": 20,
      "marginStart": 10,
      "marginTop": 100,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
    testID="textItem"
  >
     No Data Found. Try again!
  </Text>
</View>
`;
